src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Apprentice/ConfirmEditApprenticehsipRequestToConfirmEditViewModelMapper.cs

```diff
@@ -0,0 +1,112 @@
+            {
+                courseDetails = courseDetailsOriginal;
+            }
+            else
+            {
+                courseDetails = await _commitmentApi.GetTrainingProgramme(source.CourseCode);
+            }
+
+            var vm = new ConfirmEditApprenticeshipViewModel
+            {
+                AccountHashedId = source.AccountHashedId,
+                ApprenticeshipHashedId = source.HashedApprenticeshipId,
+                OriginalApprenticeship = new ConfirmEditApprenticeshipViewModel()
+                {
+                    ULN = apprenticeship.Uln
+                }
+            };
+
+            if (source.FirstName != apprenticeship.FirstName || source.LastName != apprenticeship.LastName)
+            {
+                vm.FirstName = source.FirstName;
+                vm.LastName = source.LastName;
+            }
+
+            vm.OriginalApprenticeship.FirstName = apprenticeship.FirstName;
+            vm.OriginalApprenticeship.LastName = apprenticeship.LastName;
+
+            if (source.DateOfBirth.Date != apprenticeship.DateOfBirth)
+            {
+                vm.BirthDay =   source.BirthDay;
+                vm.BirthMonth = source.BirthMonth;
+                vm.BirthYear =  source.BirthYear;
+            }
+            vm.OriginalApprenticeship.BirthDay = apprenticeship.DateOfBirth.Day;
+            vm.OriginalApprenticeship.BirthMonth = apprenticeship.DateOfBirth.Month;
+            vm.OriginalApprenticeship.BirthYear = apprenticeship.DateOfBirth.Year;
+
+            if (source.Cost != currentPrice)
+            {
+                vm.Cost = source.Cost;
+            }
+            vm.OriginalApprenticeship.Cost = currentPrice;
+            vm.EmployerReference = source.EmployerReference;
+            vm.OriginalApprenticeship.EmployerReference = apprenticeship.EmployerReference;
+
+            if (source.StartDate.Date != apprenticeship.StartDate)
+            {
+                vm.StartMonth = source.StartMonth;
+                vm.StartYear = source.StartYear;
+            }
+            vm.OriginalApprenticeship.StartMonth = apprenticeship.StartDate.Month;
+            vm.OriginalApprenticeship.StartYear = apprenticeship.StartDate.Year;
+
+            if (source.EndDate.Date != apprenticeship.EndDate)
+            {
+                vm.EndMonth = source.EndMonth;
+                vm.EndYear = source.EndYear;
+            }
+            vm.OriginalApprenticeship.EndMonth = apprenticeship.EndDate.Month;
+            vm.OriginalApprenticeship.EndYear = apprenticeship.EndDate.Year;
+
+            if (source.CourseCode != apprenticeship.CourseCode)
+            {
+                vm.CourseCode = source.CourseCode;
+                vm.TrainingName = courseDetails?.TrainingProgramme.Name;
+            }
+            vm.OriginalApprenticeship.CourseCode = apprenticeship.CourseCode;
+            vm.OriginalApprenticeship.TrainingName = courseDetailsOriginal?.TrainingProgramme.Name;
```

Comment:
could this not be apprenticeship.CourseName?

If this change is made, then the api call above on line 36 could be moved within the if statement.

Reply:
made changes for this